src/tree/mod.rs

```diff
@@ -1134,6 +1248,10 @@ impl<T> Tree<T> {
     }
 }
 
+trait IteratorNew<'a, T, I> {
+    fn new(tree: &'a Tree<T>, node_id: NodeId) -> I;
+}
+
```

Comment:
I would try if making them non-public "normal" methods does compile and otherwise just revert back to this solution.